java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java

```diff
@@ -414,75 +405,126 @@ private static void processYamlFile(File yamlFile, ReaderState state,
+                    nothingName = "EFI_ADC_NONE";
+                    break;
+                default:
+                    pinType = "brain_pin_e";
+                    nothingName = "GPIO_UNASSIGNED";
+            }
+            Map<String, Value> enumList = state.enumsReader.getEnums().get(pinType);
+            for (Map.Entry<String, Value> kv : enumList.entrySet()){
+                if (kv.getKey().equals(listPins.get(i).get("id"))){
+                    int index = kv.getValue().getIntValue();
+                    classList.ensureCapacity(index + 1);
+                    for (int ii = classList.size(); ii <= index; ii++) {
+                        classList.add(null);
+                    }
+                    classList.set(index, (String) listPins.get(i).get("ts_name"));
+                    break;
+                }
             }
         }
-        registry.register(outputEnumName, sb.toString());
-    }
-
-    private static void findMatchingEnum(String id, String ts_name, String className, ReaderState state,
-                                      Map<Integer, String> listOutputs,
-                                      Map<Integer, String> listAnalogInputs,
-                                      Map<Integer, String> listEventInputs,
-                                      Map<Integer, String> listSwitchInputs) {
-        Objects.requireNonNull(id, "id");
-        Objects.requireNonNull(className, "classname for " + id);
-
-        switch (className) {
-            case "outputs":
-                assignPinName("brain_pin_e", ts_name, id, listOutputs, state, "GPIO_UNASSIGNED");
-                break;
-            case "analog_inputs":
-                assignPinName("adc_channel_e", ts_name, id, listAnalogInputs, state, "EFI_ADC_NONE");
-                break;
-            case "event_inputs":
-                assignPinName("brain_pin_e", ts_name, id, listEventInputs, state, "GPIO_UNASSIGNED");
-                break;
-            case "switch_inputs":
-                assignPinName("brain_pin_e", ts_name, id, listSwitchInputs, state, "GPIO_UNASSIGNED");
-                break;
-        }
-    }
-
-    private static void assignPinName(String enumName, String ts_name, String id, Map<Integer, String> list, ReaderState state, String nothingName) {
-        Map<String, Value> enumList = state.enumsReader.getEnums().get(enumName);
-        for(Map.Entry<String, Value> kv : enumList.entrySet()){
-            if(kv.getKey().equals(id)){
-                list.put(kv.getValue().getIntValue(), ts_name);
-            } else if (kv.getKey().equals(nothingName)) {
-                list.put(kv.getValue().getIntValue(), "NONE");
+        for (Map.Entry<String, ArrayList<String>> kv : names.entrySet()) {
+            String outputEnumName = "";
+            String pinType = "";
+            String nothingName = "";
+            switch (kv.getKey()) {
+                case "outputs":
+                    outputEnumName = "output_pin_e_enum";
```

Comment:
I've just pushed https://github.com/rusefi/rusefi/blob/master/java_tools/configuration_definition/src/com/rusefi/PinType.java

please refactor that enum to your liking

open question if find() method should throw exception or return null for unexpected? at the moment I am not sure why do we not handle "else" as a failure

Reply:
oh nice. I will use this. tomorrow.